hack-sound-server

```diff
@@ -0,0 +1,158 @@
+        self._dbus_id = None
+        self.metadata = metadata
+        self.hold()
+        self.players = {}
+
+    def do_dbus_register(self, connection, path):
+        info = Gio.DBusNodeInfo.new_for_xml(self._DBUS_XML)
+        self._dbus_id = connection.register_object(path,
+                                                   info.interfaces[0],
+                                                   self.__method_called_cb)
+        return Gio.Application.do_dbus_register(self, connection, path)
+
+    def do_dbus_unregister(self, connection, path):
+        Gio.Application.do_dbus_unregister(self, connection, path)
+        if not self._dbus_id:
+            return
+        connection.unregister_object(self._dbus_id)
+        self._dbus_id = None
+
+    def play_sound(self, sound_event_id, connection, sender, path, iface,
+                   invocation):
+        if sound_event_id not in self.metadata:
+            invocation.return_dbus_error(self._DBUS_UNKNOWN_SOUND_EVENT_ID,
+                "sound event with id %s does not exist" % sound_event_id)
+            return
+
+        uuid_ = str(uuid.uuid4())
+        self.players[uuid_] = HackSoundPlayer(self.metadata[sound_event_id],
+                                              sender)
+        self.players[uuid_].connect("eos", self.__player_eos_cb, uuid_)
+        self.players[uuid_].connect("error", self.__player_error_cb, uuid_,
+                                    connection, path, iface)
+        self.players[uuid_].play()
+        invocation.return_value(GLib.Variant('(s)', (uuid_, )))
+
+    def __method_called_cb(self, connection, sender, path, iface,
+                           method, params, invocation):
+        if method == "PlaySound":
+            self.play_sound(params[0], connection, sender, path, iface,
+                            invocation)
+
+    def __player_eos_cb(self, unused_player, uuid_):
+        del self.players[uuid_]
+
+    def __player_error_cb(self, player, error, debug, uuid_, connection,
+                          path, iface):
+        data = (uuid_, error.message, error.domain, error.code, debug)
+        vdata = GLib.Variant("(sssis)", data)
+        if uuid_ in self.players:
+            del self.players[uuid_]
+            connection.emit_signal(None, path, iface, 'Error', vdata)
+
+
+if __name__ == "__main__":
+    metadata_path = get_metadata_path()
+    if not os.path.exists(metadata_path):
+        _logger.error("The metadata file at '%s' does not exist" % metadata_path)
+        sys.exit(1)
+
+    with open(get_metadata_path(), "r") as metadata_file:
+        # TODO
+        # Check valid JSON schema.
+        metadata = json.load(metadata_file)
+        Gst.init(None)
+        server = HackSoundServer(metadata)
+        GLib.unix_signal_add(GLib.PRIORITY_DEFAULT, signal.SIGINT, server.quit)
+        GLib.unix_signal_add(GLib.PRIORITY_DEFAULT, signal.SIGTERM, server.release)
+        server.run(None)
```

Comment:
Probably best to put `server.run(sys.argv)` here, in case we decide to support argv in the future.